libraries/lib-ipc/internal/BufferedIPCChannel.cpp

```diff
@@ -0,0 +1,158 @@
+#include "BufferedIPCChannel.h"
+
+#include <cassert>
+#include <cstring>
+#include <mutex>
+#include "MemoryX.h"
+#include "socket_guard.h"
+
+BufferedIPCChannel::BufferedIPCChannel()
+{
+   mOutputBuffer.reserve(DefaultOutputBufferCapacity);
+}
+
+BufferedIPCChannel::~BufferedIPCChannel()
+{
+   if(mSocket != INVALID_SOCKET)
+      //interrupts any blocking calls
+      CLOSE_SOCKET(mSocket);
```

Comment:
Good.  Does the operating system also give us any guarantees when the process holding one side of the socket dies abnormally?  I’m sure it should, but please point me to a reference.  I think we rely on that, so Audacity does not hang if PluginHost chokes. — Have we actually tested with a corrupt plugin?  Now that I can build swh Ladspa for Putin’s fro lm source, I could easily modify the source to make bad plug-ins.

Reply:
>Have we actually tested with a corrupt plugin?

I did test it, though on windows only.

>Does the operating system also give us any guarantees when the process holding one side of the socket dies abnormally?

For [posix-compliant](https://pubs.opengroup.org/onlinepubs/9699919799/functions/_Exit.html#tag_16_01) exit:
```
Process termination caused by any reason shall have the following consequences:
...
All of the file descriptors, directory streams, conversion descriptors, and message catalog descriptors open in the calling process shall be closed.
```